src/contract-details/ContractDetailsCard.js

```diff
@@ -48,24 +49,25 @@ export default class ContractDetailsCard extends Component {
 						</tr>
 					</tbody>
 				</table>
-				{(validToSell && !validationError) ?
-					<button
-						className="sell-at-market-btn"
-						onClick={() => actions.sellContract(contract.contract_id, 0)}
-					>
-						Sell at Market (<NumberPlain
-							value={contract.bid_price}
-							currency={contract.currency}
-							isProfit={v => v - contract.buy_price}
-						/>)
-					</button> :
+				{(durationUnit !== 't') && (
```

Comment:
So if durationUnit is of tick value then , is_valid_to_sell would be 0 , is that what you are insinuating ? If that's the case i think its from the backend perhaps. The (durationUnit !== 't') is to check explicitly, with that i don't think sell on market would appear not even for a second .